spectrocloud/resource_cluster_group.go

```diff
@@ -0,0 +1,305 @@
+package spectrocloud
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/spectrocloud/hapi/models"
+	"github.com/spectrocloud/terraform-provider-spectrocloud/pkg/client"
+)
+
+func resourceClusterGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceClusterGroupCreate,
+		ReadContext:   resourceClusterGroupRead,
+		UpdateContext: resourceClusterGroupUpdate,
+		DeleteContext: resourceClusterGroupDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		SchemaVersion: 2,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the cluster group",
+			},
+			"context": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "tenant",
+				ValidateFunc: validation.StringInSlice([]string{"", "project", "tenant"}, false),
+				Description:  "Cluster group context can be 'project' or 'tenant'. Defaults to 'project'.",
+			},
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      schema.HashString,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "A list of tags to be applied to the cluster group. Tags must be in the form of `key:value`.",
+			},
+			"config": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host_endpoint_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "Ingress",
+							ValidateFunc: validation.StringInSlice([]string{"", "Ingress", "LoadBalancer"}, false),
```

Comment:
We can remove this empty string in validation list, since it a optional attribute by default it will `""`, in its not in config

Reply:
```suggestion
							ValidateFunc: validation.StringInSlice([]string{"Ingress", "LoadBalancer"}, false),
```